### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by sfirrin**
### What does this PR do?

Adds snapshot integration tests for the layer. 

Includes four sample Lambdas:
* one that sends metrics async
* one that sends sync metrics
* one that generates Datadog traces
* one that makes HTTP requests (to test patching of HTTP calls)

I'm copy/pasting [this wiki entry](https://github.com/DataDog/devops/wiki/Lambda-Layers#integration-tests) about these tests below:

## Integration tests

These tests produce snapshots of the return value, logging, and HTTP output of sample Lambda functions deployed with the layer so that you can see exactly what has changed with any change to the layer.

These are currently implemented in the Python and Node layers.

### Usage

Find API key from 1password named _Lambda Layer Integrations Test API Key_.

To run integration tests for new changes to the layer:

```bash
BUILD_LAYERS=true DD_API_KEY=<api key> aws-vault exec sandbox-account-admin -- ./scripts/run_integration_tests.sh
```

If the new change to the layer code causes a change to the output of any of our sample Lambda functions then the integration test will fail and print what has changed in the output between this execution and the snapshot:
```
Failed: Mismatch found between new http-requests_python36 logs (first) and snapshot (second):
2c2
< {"m": "aws.lambda.enhanced.invocations", "v": 1, "e": XXXX, "t": ["region:us-east-1", "account_id:XXXX", "functionname:integration-tester-dev-http-requests_python36", "new_tag:new_value", "cold_start:true", "memorysize:1024", "runtime:python3.6", "dd_lambda_layer:datadog-python36_2.XX.0"]}
---
> {"m": "aws.lambda.enhanced.invocations", "v": 1, "e": XXXX, "t": ["region:us-east-1", "account_id:XXXX", "functionname:integration-tester-dev-http-requests_python36", "cold_start:true", "memorysize:1024", "runtime:python3.6", "dd_lambda_layer:datadog-python36_2.XX.0"]}
```
If the change that is printed is what you expect, you'll want to write new snapshots that reflect your expected change. 

To generate new snapshots:
```bash
UPDATE_SNAPSHOTS=true DD_API_KEY=<api key> aws-vault exec sandbox-account-admin -- ./scripts/run_integration_tests.sh
```

All changes made to the snapshots will be included in your PR, so that the changes to the behavior of the layer will be easy to see alongside the changes to the code.

### Implementation

These integration tests do the following:

1. Build layers using the `./scripts/build_layers.sh` script (or don't if `$BUILD_LAYERS` isn't set)
1. Deploy sample Lambda handlers for all runtimes using the serverless framework, setting the env var `DD_INTEGRATION_TEST` to true
1. If `DD_INTEGRATION_TEST` is true then the layer will log all patched requests in the format `HTTP GET https://ip-ranges.datadoghq.com/ Headers: ["x-datadog-parent-id:abc", "x-datadog-sampling-priority:2", "x-datadog-trace-id:xyz"] Data: {}`
1. Invoke each Lambda once for each of the event types defined in `./input_events`
1. Compare the return value of each Lambda invocation for each input event with the snapshot that we have saved for the input event in `./snapshots/return_values`
1. Query for the logs of each Lambda function
1. Replace invocation-specific data (like timestamps and request IDs) with XXXX to normalize log snapshots
1. Compare new logs to the snapshot log file that we have saved in `./tests/integration/snapshots/logs`
1. Print any mismatches between snapshots and new logs / return values
1. Fail the script if a mismatch is found, otherwise succeed

See [the script for the Python layer](https://github.com/DataDog/datadog-lambda-layer-python/blob/master/scripts/run_integration_tests.sh).

### CODE REVIEW

File: scripts/publish_prod.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ fi
 
 yarn login
 
-./scripts/run_tests.sh
+# ./scripts/run_tests.sh
```
Comment (DarcyRaynerDD) [inline scripts/publish_prod.sh]: I still can't figure out why this script is breaking.

Comment (sfirrin) [reply scripts/publish_prod.sh]: oops didn't mean to commit this

File: src/trace/patch-http.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ function patchMethod(mod: typeof http | typeof https, method: "get" | "request",
       const { options, callback } = normalizeArgs(arg1, arg2, arg3);
       const requestOpts = getRequestOptionsWithTraceContext(options, contextService);
 
+      if (isIntegrationTest()) {
```
[the remark this answers is not in the document]
Comment (sfirrin) [reply src/trace/patch-http.ts]: agreed

File: src/trace/patch-http.ts [hunk 3]
```diff
@@ -87,3 +91,36 @@ function getRequestOptionsWithTraceContext(
...
   };
 }
+
+function isIntegrationTest() {
+  const integrationTestEnvVar = process.env.DD_INTEGRATION_TEST;
+  if (typeof integrationTestEnvVar !== "string") {
+    return false;
+  }
+
+  if (integrationTestEnvVar.toLowerCase() === "true") {
```
[the remark this answers is not in the document]
Comment (sfirrin) [reply src/trace/patch-http.ts]: :+1:

**Pull request merged**